jee-commons/src/main/java/biz/paluch/jee/commons/BeanLookup.java

```diff
@@ -136,5 +153,14 @@ public static BeanManager beanManager() {
     public static void reset() {
         lookupStrategy = new BeanManagerLookupStrategy();
     }
+    
+    private static BeanManager beanManagerLookup() {
+        for(String jndiName : BEANMANAGER_JNDI_NAMES) {
+            BeanManager beanManager = NamingLookup.doLookup(jndiName);
```

Comment:
NamingLookup should throw an exception. I'd propose to surround the lookup with a try/catch and fetching the first exception occured. You can then use that exception in case no bean manager was found, like throw new ...Exception("Could not find any BeanManager from " + BEANMANAGER_JNDI_NAMES, rootCause);